File: LeetCode/cpp/19.cpp
class Solution {
public:
    ListNode *removeNthFromEnd(ListNode *head, int n) {
        ListNode *l = new ListNode(0);
        l->next = head;
        ListNode *fast = head, *slow = head;
        ListNode *prev = l;
        while (n > 0) {
            fast = fast->next;
            --n;
        }
        while (fast) {
            fast = fast->next;
            prev = slow;
            slow = slow->next;
        }
        prev->next = slow->next;
        return l->next;
    }
};

File: LeetCode/python3/19.py
class Solution:
    def removeNthFromEnd(self, head: ListNode, n: int) -> ListNode:
        i = 0
        fast, slow = head,head
        while i < n:
            fast = fast.next
            i += 1
        if not fast:
            return slow.next
        while fast and fast.next:
            fast = fast.next
            slow = slow.next
        slow.next = slow.next.next if slow.next else None
        return head
